tests/integration/cloud/helpers/cloud_test_base.py

```diff
@@ -0,0 +1,175 @@
+        return self.__instance_name
+
+    @property
+    def providers(self):
+        if not hasattr(self, '__providers'):
+            self.__providers = self.run_cloud('--list-providers')
+        return self.__providers
+
+    @property
+    def provider_config(self):
+        if not hasattr(self, '__provider_config'):
+            self.__provider_config = cloud_providers_config(
+                path.join(
+                    FILES,
+                    'conf',
+                    'cloud.providers.d',
+                    self.PROVIDER + '.conf'
+                )
+            )
+        return self.__provider_config[self.profile_str][self.PROVIDER]
+
+    @property
+    def config(self):
+        if not hasattr(self, '__config'):
+            self.__config = cloud_config(
+                path.join(
+                    FILES,
+                    'conf',
+                    'cloud.profiles.d',
+                    self.PROVIDER + '.conf'
+                )
+            )
+        return self.__config
+
+    @property
+    def profile_str(self):
+        return self.PROVIDER + '-config'
+
+    @expensiveTest
+    def setUp(self):
+        '''
+        Sets up the test requirements.  In child classes, define PROVIDER and REQUIRED_CONFIG_ITEMS or this will fail
+        '''
+        super(CloudTest, self).setUp()
+
+        if not self.PROVIDER:
+            self.fail('A PROVIDER must be defined for this test')
+
+        # check if appropriate cloud provider and profile files are present
+        if self.profile_str + ':' not in self.providers:
+            self.skipTest(
+                'Configuration file for {0} was not found. Check {0}.conf files '
+                'in tests/integration/files/conf/cloud.*.d/ to run these tests.'
+                .format(self.PROVIDER)
+            )
+
+        missing_conf_item = []
+        for att in self.REQUIRED_PROVIDER_CONFIG_ITEMS:
+            if not self.provider_config.get(att):
+                missing_conf_item.append(att)
+
+        if missing_conf_item:
+            self.skipTest('Conf items are missing that must be provided to run these tests:  {}'
+                          .format(', '.join(missing_conf_item)) +
+                          '\nCheck tests/integration/files/conf/cloud.providers.d/{0}.conf'.format(self.PROVIDER))
+
+        self.assertFalse(self._instance_exists(),
+                         'The instance "{}" exists before it was created by the test'.format(self.instance_name))
+
+    def tearDown(self):
+        '''
+        Clean up after tests, If the instance still exists for any reason, delete it.
+        Instances should be destroyed before the tearDown, _destroy_instance() should be called exactly
+        one time in a test for each instance created.  This is a failSafe and something went wrong
+        if the tearDown is where an instance is destroyed.
+        '''
+        instance_deleted = True
+        tries = 0
```

Comment:
- Why define tries here? It'll always start a 0 when calling range.
- We might consider making the number of tries and the sleep time variables in case we want to tweak them in the future.

Reply:
That was so that I could include it in a test failure message outside of the for loop